README.md

```diff
@@ -1,2 +1,21 @@
 # comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+a) This algorithm uses the length of list – 1 for i and j and then asks if i is not equal to j and list[i] is equal to list[j] then True is returned otherwise False is returned. So essentially it asks if two items in the same list are the same then true is returned. If they aren’t the same then we get False.
+
+b) Because i and j are both equal to 0 to n-1, so the algorithm takes n^2 operations. Which means that if the size of the list is doubled then the time taken to go through this algorithm will quadruple meaning that it is quadratic.
+
+c) The algorithm is still correct because i is equal to 0 to n – 1 anyway. 
+
+d) It will run twice as fast because we’re incrementing j at the same time as i rather than incrementing them separately. 
```

Comment:
The reason it's faster is that roughly half of the loop iterations are eliminated -- instead of checking each pair twice as both `i,j` and `j,i`, only one is checked.